File: CSCD212/CommandPattern/src/DocumentEraseCommand.java
/** A command designed to erase some text from the document. */
public class DocumentEraseCommand implements Command {
    private Document document;
    private String   text;

    /**
     * @param document Document object passed from the DocumentInvoker class.
     * @param text     Text to be erased from the document.
     */
    public DocumentEraseCommand(Document document, String text) {
        this.document = document;
        this.text     = text;
        document.Erase(text);
    }

    @Override
    public void undo() {
        document.Write(text);
    }

    @Override
    public void redo() {
        document.Erase(text);
    }
}

File: Personal/CommandPattern/pkg/document_erase_command.py
from pkg.command import Command


class DocumentEraseCommand(Command):
    """Remove 'text' from the document."""

    def __init__(self, document, text):
        """Constructor.

        Parameters
        ----------
        document : document.Document
            Document object/instance.
        text : str
            Text to be added to the document.
        """
        self._document = document
        self._text = text
        self._document.erase(text)

    def undo(self):
        """Add 'text' back to the document."""
        self._document.write(self._text)

    def redo(self):
        """Re-remove 'text' to the document."""
        self._document.erase(self._text)
